services/github/github-issues-search.tester.js

```diff
@@ -0,0 +1,50 @@
+'use strict'
+
+const { isMetric } = require('../test-validators')
+const { ServiceTester } = require('../tester')
+const t = (module.exports = new ServiceTester({
+  id: 'GithubIssuesSearch',
+  title: 'Github Issues Search',
+  pathPrefix: '/github',
+}))
+
+t.create('GitHub issue search (valid query string)')
+  .get(
+    '/issues-search.json?query=repo%3Abadges%2Fshields%20is%3Aclosed%20label%3Ablocker%20'
+  )
+  .expectBadge({
+    label: 'query',
+    message: isMetric,
+  })
+
+t.create('GitHub issue search (invalid query string)')
+  .get('/issues-search.json?query=')
+  .expectBadge({
+    label: 'query',
+    message: 'invalid query parameter: query',
+  })
+
+t.create('GitHub Repo issue search (valid query string)')
+  .get(
+    '/issues-search/badges/shields.json?query=is%3Aclosed%20label%3Ablocker%20'
+  )
+  .expectBadge({
+    label: 'query',
+    message: isMetric,
+  })
+
+t.create('GitHub Repo issue search (invalid query string)')
+  .get('/issues-search/badges/shields.json?query=')
+  .expectBadge({
+    label: 'query',
+    message: 'invalid query parameter: query',
+  })
+
+t.create('GitHub Repo issue search (invalid repo)')
+  .get(
+    '/issues-search/badges/helmets.json?query=is%3Aclosed%20label%3Ablocker%20'
+  )
+  .expectBadge({
+    label: 'query',
+    message: '0',
```

Comment:
One thing I hadn't focussed on when I originally proposed this is that if we use the `search` API instead of the `repository` API, passing a repo that doesn't exist just returns zero search results (same as passing a username or label that doesn't exist, or a text string that isn't found in any issue). I think that's OK for a custom query endpoint, but I think that changes the tradeoff when it comes to making all our other issue badges a special-case of this one. We'd also have to accept worse error handling, so maybe we just leave them as they are.

Reply:
Can't remember which service for the life of me but feel like we had a recent-ish new service that managed to hook into some extra node/field that gave more contextual error information.

Probably not the end of the world from an error message, but if it is problematic not being able to differentiate the repo not found vs found but zero results then we could consider firing off a different call in parallel to check whether the repo exists and use that in the error message (or only do the check in the case of zero results)